src/pages/ThorChainLP/YourPositions.tsx

```diff
@@ -182,87 +66,209 @@ export const YourPositions = () => {
+        display: { base: 'none', lg: 'table-cell' },
+        Cell: ({ row }: { value: string | undefined; row: RowProps }) => {
+          const position = row.original
+          const pool = useMemo(
+            () => pools?.find(pool => pool.assetId === position.assetId),
+            [position.assetId],
+          )
+          const poolAssetId = pool?.assetId
+          const thorchainNotationPoolAssetId = pool?.asset
+          const asset = useAppSelector(state => selectAssetById(state, poolAssetId ?? ''))
 
-        return (
-          <PositionButton
-            key={position.opportunityId}
-            poolAssetId={pool.asset}
-            position={position}
-          />
-        )
-      })
-      .filter(isSome)
+          const { data: earnings, isLoading: isEarningsLoading } = useQuery({
+            ...reactQueries.thorchainLp.earnings(position.dateFirstAdded),
+            // @lukemorales/query-key-factory only returns queryFn and queryKey - all others will be ignored in the returned object
+            // No staleTime, meaning cache-then-fresh (or fresh if now > garbage collection time)
+            // That ensures new active listeners always get fresh earnings data
+            staleTime: 0,
+            select: data => {
+              const poolEarnings = data?.meta.pools.find(
+                pool => pool.pool === thorchainNotationPoolAssetId,
+              )
+              if (!poolEarnings) return null
+
+              return calculateEarnings(poolEarnings, position.poolShare, runeMarketData.price)
+            },
+          })
+
+          if (!asset) return null
+
+          return (
+            <Stack display={mobileDisplay} spacing={0}>
+              <Skeleton isLoaded={!isEarningsLoading}>
+                <Amount.Fiat value={earnings?.totalEarningsFiat ?? '0'} />
+              </Skeleton>
+              <Skeleton isLoaded={!isEarningsLoading}>
+                <Amount.Crypto
+                  value={earnings?.assetEarningsCryptoPrecision ?? '0'}
+                  symbol={asset.symbol}
+                  fontSize='sm'
+                  color='text.subtle'
+                />
+              </Skeleton>
+              <Skeleton isLoaded={!isEarningsLoading}>
+                <Amount.Crypto
+                  value={earnings?.runeEarningsCryptoPrecision ?? '0'}
+                  symbol={'RUNE'}
+                  fontSize='sm'
+                  color='text.subtle'
+                />
+              </Skeleton>
+            </Stack>
+          )
+        },
+      },
+      {
+        Header: translate('pools.poolShare'),
+        accessor: 'poolOwnershipPercentage',
+        display: { base: 'none', lg: 'table-cell' },
+        Cell: ({ row }: { value: string | undefined; row: RowProps }) => {
+          const position = row.original
+
+          return (
+            <Box display={largeDisplay}>
+              <Amount.Percent
+                options={{ maximumFractionDigits: 8 }}
```

Comment:
can be moved to top of file

Reply:
Not sure what do we mean here, moving this object above in an effort to avoid re-renders? This whole expression is already memoized